src/net_processing.cpp

```diff
@@ -3417,6 +3447,274 @@ bool ProcessMessage(CNode* pfrom, const std::string& msg_type, CDataStream& vRec
         return true;
     }
 
+    std::chrono::microseconds current_time = GetTime<std::chrono::microseconds>();
+
+    // Record an (expected) reconciliation request with parameters to respond when time comes.
+    // All initial reconciliation responses will be done at the same time to prevent tx-related privacy leaks.
+    if (strCommand == NetMsgType::REQRECON) {
+        if (!pfrom->m_recon_state->sender) return true;
```

Comment:
One more potential issue: If I understand it correctly, `m_recon_state` is null if our peer does not signal Erlay support. So we should check if `m_recon_state` is non-null before accessing its members here.

Otherwise an attacker could not signal Erlay but then send Erlay-specific messages to crash our node. Same applies for the other new messages `NetMsgType::SKETCH` etc. below.

Reply:
Definitely an issue, will address!
I had asserts all over, and then was replacing them with returns, but this one requires 2 checks for m_recon_state and for sender :)